### PR TITLE
Fix failing tests on `main` due to torch 2.1

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix failing tests on `main` due to torch 2.1

### CODE REVIEW

File: tests/models/hubert/test_modeling_hubert.py [hunk 1]
```diff
@@ -527,25 +524,28 @@ def flatten_output(output):
...
-                loaded_output = flatten_output(loaded_output)
-
-                for i in range(num_outputs):
-                    self.assertTrue(
-                        torch.allclose(model_output[i], loaded_output[i]),
-                        f"serialized model {i}th output doesn't match model {i}th output for {model_class}",
-                    )
+            # This part fails since torch 2.1 (for wav2vec2 and hubert):
+            # (`Couldn't serialize / deserialize the traced model: Serialization of parametrized modules is only supported through state_dict()`)
+            # TODO: Fix it.
```
Comment (ydshieh) [inline tests/models/hubert/test_modeling_hubert.py:529]: despite the changes in the modeling files, this part still fails

File: src/transformers/models/hubert/modeling_hubert.py [hunk 2]
```diff
@@ -490,17 +491,19 @@ def forward(
...
 
-        if attn_weights.size() != (bsz * self.num_heads, tgt_len, src_len):
-            raise ValueError(
-                f"Attention weights should be of size {(bsz * self.num_heads, tgt_len, src_len)}, but is"
-                f" {attn_weights.size()}"
-            )
-
-        if attention_mask is not None:
-            if attention_mask.size() != (bsz, 1, tgt_len, src_len):
+        if not is_torch_fx_proxy(attn_weights):
```
Comment (ArthurZucker) [inline src/transformers/models/hubert/modeling_hubert.py:494]: let's do the check a lot earlier, we can check this in the model's forward only once!

**Review by ArthurZucker — APPROVED** [review]
Thanks, let's try to use unitest skip and put the torch fx proxy in the model forward

**Review by LysandreJik — APPROVED** [review]
LGTM, let's ensure we check for eventual performance issues and fix the TODOs quickly.

Thanks for fixing these tests so quickly!

File: src/transformers/models/hubert/modeling_hubert.py (hunk 2 above)
Comment (ydshieh) [reply src/transformers/models/hubert/modeling_hubert.py:494]: I am not sure about this place. This block contain

```
            attn_weights = attn_weights.view(bsz, self.num_heads, tgt_len, src_len) + attention_mask
            attn_weights = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
```
and it should be run no matter if we are in fx proxy or not

**Review by ArthurZucker — APPROVED** [review]
Thanks

**Comment by ydshieh** [comment]
@michaelbenayoun Could you help us on the torch fx tests for wav2vec2/hubert with torch 2.1.

See [this internal discussion](https://huggingface.slack.com/archives/C01NE71C4F7/p1695971833833119)

But in short, it can't do
```
if attn_weights.size() != (bsz * self.num_heads, tgt_len, src_len)
```
as the corresponding proxy object has no `_metadata` attribute (with torch 2.1) but it has it with torch 2.0.

**Pull request merged**